src/actions/actions.ts

```diff
@@ -5287,3 +5288,287 @@ class ActionOverrideCmdAltUp extends BaseCommand {
+
+      // Search all occurences of the character pressed after the cursor
+      if (searchChar === " ") { // Searching for space should only find the first space
+        return vimState.easyMotion.sortedSearch(position, " {1,}", {
+          isRegex: true,
+          max: position
+        });
+      } else {
+        return vimState.easyMotion.sortedSearch(position, searchChar, {
+          max: position
+        });
+      }
+    }
+
+    public getMatchPosition(match: EasyMotion.Match, position: Position, vimState: VimState): Position {
+        return new Position(match.position.line, Math.max(0, match.position.character + 1));
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionWordCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "w"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words after the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        min: position
+      });
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionEndForwardCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "e"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the end of all words after the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        min: position
+      });
+    }
+
+    public getMatchPosition(match: EasyMotion.Match, position: Position, vimState: VimState): Position {
+        return new Position(match.position.line, match.position.character + match.text.length - 1);
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionEndBackwardCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "g", "e"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words before the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        max: position,
+      });
+    }
+
+    public getMatchPosition(match: EasyMotion.Match, position: Position, vimState: VimState): Position {
+        return new Position(match.position.line, match.position.character + match.text.length - 1);
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionBeginningWordCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "b"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words before the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        max: position,
+      });
+    }
+  }
+
+  @RegisterAction
+  class CommandEscEasyMotionMode extends BaseCommand {
```

Comment:
I actually disagree

Originally I was going to make the same comment, however I then thought this was more modular if/when he moves it to its own file

Why do we have a Replace vs VisualReplace sort of thing, behavior is different so its a different action